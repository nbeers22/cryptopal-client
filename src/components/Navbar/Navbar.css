nav{
  background-color: #333333;
}
nav .container{
  display: flex;
}
nav img{
  height: 30px;
  width: 30px;
  border-radius: 50%;
}
.top-bar-left, .top-bar-right{
  width: 80%;
  align-items: center;
}
.top-bar-right{
  width: 20%;
  min-width: 200px;
  position: relative;
}
.top-bar-right ul{
  list-style-type: none;
  padding-left: 0;
  text-align: right;
}
.top-bar-right li a{
  padding: 10px 20px;
  color: #FFF;
  text-decoration: none;
}
.name{
  position: relative;
  bottom: 10px;
}
li {
  display: block;
  transition-duration: 0.5s;
}
li:hover {
  cursor: pointer;
}
ul li ul {
  visibility: hidden;
  opacity: 0;
  position: absolute;
  transition: all 0.5s ease;
  padding-top: 1rem;
  display: none;
  width: 130px;
  right: 0;
}

ul li:hover > ul,
ul li:focus-within > ul {
  visibility: visible;
  opacity: 1;
  display: block;
  background-color: #333333;
}
ul li ul li {
  clear: both;
  width: 100%;
  text-align: left;
}
ul li ul li a{
  display: block;
  transition: all 0.25s ease-in-out;
}
ul li ul li a:hover{
  background-color: #FFF;
  color: #333;
}
li:focus-within a {
  outline: none;
}