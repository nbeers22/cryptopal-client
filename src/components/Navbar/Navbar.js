import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCaretDown } from '@fortawesome/free-solid-svg-icons'
import './Navbar.css';
import TokenService from '../../services/token-service';

function Navbar(props) {
  let navbarItems;
  const gravatarURL = TokenService.getAuthGravatar();
  const userName = TokenService.getAuthName();

  if(window.sessionStorage.getItem('cryptopal-client-auth-token')){
    navbarItems = 
      <ul>
        <li><a href="#" aria-haspopup="true"><img src={ gravatarURL } alt={ userName } /> <span className="name">{ userName } <FontAwesomeIcon icon={faCaretDown} /></span></a>
          <ul className="dropdown" aria-label="submenu">
            <li><Link to="/dashboard">Dashboard</Link></li>
            <li><Link to="/account">Edit Account</Link></li>
            <li><Link to="/logout" onClick={logout}>Logout</Link></li>
          </ul>
        </li>
      </ul>
  }else{
    navbarItems = 
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/signup">Sign up</Link></li>
        <li><Link to="/login">Sign in</Link></li>
      </ul>
  }

  function logout(event){
    event.preventDefault();
    TokenService.clearAuthData();
    props.history.push('/')
  }

  return(
    <nav role="navigation">
      <div className="container">
        <div className="top-bar-left">
          <Link to="/"><img src="" alt="Logo"/></Link>
        </div>
        <div className="top-bar-right">
          {navbarItems}
        </div>
      </div>
    </nav>
  )
}

export default withRouter(Navbar);