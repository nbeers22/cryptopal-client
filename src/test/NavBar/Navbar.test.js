import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer';
import ReactRouterEnzymeContext from 'react-router-enzyme-context'
import Navbar from '../../components/Navbar/Navbar.js'

describe('Navbar Component', () => {
  let wrapper;

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><Navbar /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the proper "not logged in" links when no user is logged in', () => {
    wrapper = shallow(<Navbar.WrappedComponent />)
    expect(wrapper.find('.menu-item').length).toBe(3)
    expect(wrapper.find('Link').find({to: '/coins'}).length).toEqual(1)
    expect(wrapper.find('Link').find({to: '/signup'}).length).toEqual(1)
    expect(wrapper.find('Link').find({to: '/login'}).length).toEqual(1)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the proper "logged in" links when user is logged in', () => {
    window.sessionStorage.setItem('cryptopal-client-auth-token', 'abc123')
    wrapper = shallow(<Navbar.WrappedComponent />)
    expect(wrapper.find('.dropdown').length).toBe(1)
    expect(wrapper.find('.menu-item').length).toBe(4)
    expect(wrapper.find('Link').find({to: '/coins'}).length).toEqual(1)
    expect(wrapper.find('Link').find({to: '/dashboard'}).length).toEqual(1)
    expect(wrapper.find('Link').find({to: '/account'}).length).toEqual(1)
    expect(wrapper.find('Link').find({to: '/logout'}).length).toEqual(1)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
