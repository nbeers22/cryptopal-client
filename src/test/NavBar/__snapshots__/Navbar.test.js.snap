// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders the proper "logged in" links when user is logged in 1`] = `
<nav
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="top-bar-left"
    >
      <Link
        to="/"
      >
        <img
          alt="CryptoPal Home"
          className="nav-logo"
          src="cents-logo.png"
        />
      </Link>
    </div>
    <div
      className="top-bar-right"
    >
      <ul>
        <li>
          <a
            aria-haspopup="true"
            href="/account"
          >
            <img
              alt={null}
              src={null}
            />
             
            <span
              className="name"
            >
               
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      320,
                      512,
                      Array [],
                      "f0d7",
                      "M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z",
                    ],
                    "iconName": "caret-down",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
          </a>
          <ul
            aria-label="submenu"
            className="dropdown"
          >
            <li
              className="menu-item"
            >
              <Link
                to="/coins"
              >
                All Coins
              </Link>
            </li>
            <li
              className="menu-item"
            >
              <Link
                to="/account"
              >
                Account
              </Link>
            </li>
            <li
              className="menu-item"
            >
              <Link
                to="/dashboard"
              >
                Dashboard
              </Link>
            </li>
            <li
              className="menu-item"
            >
              <Link
                onClick={[Function]}
                to="/logout"
              >
                Logout
              </Link>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Navbar Component renders the proper "not logged in" links when no user is logged in 1`] = `
<nav
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="top-bar-left"
    >
      <Link
        to="/"
      >
        <img
          alt="CryptoPal Home"
          className="nav-logo"
          src="cents-logo.png"
        />
      </Link>
    </div>
    <div
      className="top-bar-right"
    >
      <ul>
        <li
          className="menu-item"
        >
          <Link
            to="/coins"
          >
            All Coins
          </Link>
        </li>
        <li
          className="menu-item"
        >
          <Link
            to="/signup"
          >
            Sign up
          </Link>
        </li>
        <li
          className="menu-item"
        >
          <Link
            to="/login"
          >
            Sign in
          </Link>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
